transports/janus_websockets.c

```diff
@@ -924,17 +975,54 @@ static int janus_websockets_admin_callback(struct libwebsocket_context *this,
 			}
 			if(!ws_client->destroy && !g_atomic_int_get(&stopping)) {
 				janus_mutex_lock(&ws_client->mutex);
-				/* Shoot all the pending messages first */
+				/* Check if we have a pending/partial write to complete first */
+				if(ws_client->buffer && ws_client->bufoffset > 0
+						&& !ws_client->destroy && !g_atomic_int_get(&stopping)) {
+					int pending = ws_client->buflen - ws_client->bufoffset - LWS_SEND_BUFFER_POST_PADDING;
```

Comment:
`ws_client->buflen` is based on the longest response sent so far, not on the length of the current response. So `pending` can be larger than it should be.